presto-main/src/main/java/io/prestosql/metadata/FunctionArgumentDefinition.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.metadata;
+
+import java.util.Objects;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+
+public class FunctionArgumentDefinition
+{
+    private final boolean nullable;
+
+    public FunctionArgumentDefinition(boolean nullable)
+    {
+        this.nullable = nullable;
+    }
+
+    public boolean isNullable()
```

Comment:
What are the exact semantics of an argument tagged as `isNullable`? Does it mean it can take a null and otherwise it's an error? Or does it mean the function wants to be called with a null value and otherwise the function returns a null (i.e., `RETURNS NULL ON NULL INPUT` vs `CALLED ON NULL INPUT` from standard SQL semantics)